predicators/envs/spot_env.py

```diff
@@ -0,0 +1,299 @@
+"""Simple pick-place-move environment for the Boston Dynamics Spot Robot."""
+
+import json
+from pathlib import Path
+from typing import Collection, Dict, List, Optional, Set
+
+import matplotlib
+import numpy as np
+from gym.spaces import Box
+
+from predicators import utils
+from predicators.envs import BaseEnv
+from predicators.envs.pddl_env import _action_to_ground_strips_op, \
+    _create_predicate_classifier, _PDDLEnvState
+from predicators.settings import CFG
+from predicators.structs import Action, EnvironmentTask, GroundAtom, \
+    LiftedAtom, Object, Predicate, State, STRIPSOperator, Type, Variable
+
+
+class SpotEnv(BaseEnv):
+    """An environment containing tasks for a real Spot robot to execute.
+
+    Currently, the robot can move to specific 'surfaces' (e.g. tables),
+    pick objects from on top these surfaces, and then place them
+    elsewhere.
+    """
+
+    def __init__(self, use_gui: bool = True) -> None:
+        super().__init__(use_gui)
+
+        # Types
+        self._robot_type = Type("robot", [])
+        self._can_type = Type("soda_can", [])
```

Comment:
We might want to switch this to a general object type, and split into more specific ones later.

Reply:
i think let's leave it like this for now, since the predicates and operators would get more complicated if things were untyped